### ISSUE TITLE
`MsgGpsTime` implementation of `gps_time` is not checking status flags

### ISSUE THREAD

**Issue by marcomaida94** [swift-nav/libsbp#1363]
Hello, 
I have experienced some inconsistency in `MsgGpsTime`

```
{"crc":11885,"length":11,"msg_type":258,"msg_name":"MSG_GPS_TIME","payload":"AADM7g0AAAAAAAA=","preamble":85,"sender":789,"wn":0,"tow":913100,"ns_residual":0,"flags":0}
```

As you can see here, week number is zero (this data is recent) and the `flags` field is also `0`, which is supposed to communicate that the message is invalid. By reading the docs:
<img width="289" alt="Screenshot 2023-10-16 at 18 17 51" src="https://github.com/swift-nav/libsbp/assets/120410627/0977069c-6b88-441d-8510-06b8b99b2f0b">

However, the [`gps_time`](https://github.com/swift-nav/libsbp/blob/0bfe6c61bc22e370e50259d1e2e7a4fe1541e277/rust/sbp/src/messages/navigation.rs#L1890) function does not check for this flag, and when called it returns an invalid timestamp for the above case.

Should the function be fixed and return `None` if the status flag is 0?

Thank you!

**Comment by silverjam** [comment]
Hello @marcomaida94 -- we're discussing this internally, we should have a response for you soon.  Thanks for raising the issue!

**Comment by marcomaida94** [comment]
Thank you @silverjam for the quick reply !

**Comment by pcrumley** [comment]
Hi @marcomaida94 -- Thanks for bringing this issue to our attention. It should be closed by PR: #1375

**Comment by marcomaida94** [comment]
Thank you!
